### PR TITLE
new Algorithms added

### PR DESCRIPTION
**Pull request by Ashi-1**

### CODE REVIEW

File: C++/merge sort.cpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
+#include<iostream>
```
Comment (juhitiwari) [inline C++/merge sort.cpp]: We already have merge sort implementation in cpp. Please check readme for what algorithms you can implement.

File: C++/0-1 KnapSack problem by recursion.cpp [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#include<iostream>
```
Comment (juhitiwari) [inline C++/0-1 KnapSack problem by recursion.cpp:1]: please add entry for this algorithm in readme

**Review by juhitiwari — CHANGES_REQUESTED** [review]

**Comment by JustHackto** [comment]
Segmented Sieve

**Comment by juhitiwari** [comment]
@Ashi-1 there have been multiple PRs from your side regarding the same. I would have to mark other two as spam. Please make the final changes in this PR itself. Remove the merge sort file. I see you have added circular linked list, so update that in README and then we would be good to go for merging.

**Pull request merged**